From dbd8c68b7a16fece185bf1c9d7e337285e08bca2 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Tue, 15 Oct 2013 09:26:32 +0200
Subject: [PATCH] network/ifname-genrules.sh: fixed DRIVERS udev condition

DRIVERS!="?*" matches any empty string in DRIVERS
DRIVERS=="?*" matches any non-empty string in DRIVERS

So, we can't use != in the conditions
---
 modules.d/40network/ifname-genrules.sh | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/modules.d/40network/ifname-genrules.sh b/modules.d/40network/ifname-genrules.sh
index 5663ce4..1d144df 100755
--- a/modules.d/40network/ifname-genrules.sh
+++ b/modules.d/40network/ifname-genrules.sh
@@ -9,18 +9,17 @@ fi
 
     echo 'SUBSYSTEM!="net",  GOTO="ifname_end"'
     echo 'ACTION!="add", GOTO="ifname_end"'
-    echo 'DRIVERS!="?*", GOTO="ifname_end"'
     echo 'ATTR{type}!="1", GOTO="ifname_end"'
 
     for p in $(getargs ifname=); do
         parse_ifname_opts $p
-	printf 'ATTR{address}=="%s", NAME="%s", GOTO="ifname_end"\n' "$ifname_mac" "$ifname_if"
+	printf 'DRIVERS=="?*", ATTR{address}=="%s", NAME="%s", GOTO="ifname_end"\n' "$ifname_mac" "$ifname_if"
     done
 
     # Rename non named interfaces out of the way for named ones.
     for p in $(getargs ifname=); do
         parse_ifname_opts $p
-	printf 'ATTR{address}!="%s", KERNEL=="%s", NAME="rename$attr{ifindex}", GOTO="ifname_end"\n' "$ifname_mac"  "$ifname_if"
+	printf 'DRIVERS=="?*", ATTR{address}!="%s", KERNEL=="%s", NAME="rename$attr{ifindex}", GOTO="ifname_end"\n' "$ifname_mac"  "$ifname_if"
     done
 
     echo 'LABEL="ifname_end"'
